samples/sensor/grow_r502a/src/main.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2021 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+#include <drivers/uart.h>
+#include <drivers/gpio.h>
+
+#define FPS_DEV_NAME	DT_LABEL(DT_INST(0, hzgrow_grow_r502a))
+
+static bool enroll;
+static struct sensor_value fid;
+
+static void finger_match(const struct device *dev)
+{
+	struct sensor_value input;
+	int ret;
+
+	ret = sensor_attr_get(dev, SENSOR_CHAN_FINGERPRINT,
+		SENSOR_ATTR_FIND, &input);
+	if (ret != 0) {
+		printk("Sensor attr get failed %d\n", ret);
+		return;
+	}
+	printk("Matched ID : %d\n", input.val1);
+	printk("confidence : %d\n", input.val2);
+}
+static void finger_enroll(const struct device *dev)
+{
+	int ret;
+
+	ret = sensor_attr_set(dev, SENSOR_CHAN_FINGERPRINT, SENSOR_ATTR_ADD, &fid);
+
+	if (ret == 0) {
+		enroll = false;
+	}
+}
+
+static void trigger_handler(const struct device *dev,
+			    const struct sensor_trigger *trigger)
+{
+	if (enroll) {
+		finger_enroll(dev);
+	} else {
+		finger_match(dev);
+	}
+}
+
+void main(void)
+{
+	struct sensor_value val, del;
+	int ret;
+
+	const struct device *dev =  device_get_binding(FPS_DEV_NAME);
```

Comment:
Please use DEVICE_DT_GET_ANY or DEVICE_DT_GET_ONE. Make sure to read the documentation for the required error handling steps.